web/packages/teleport/src/Notifications/Notifications.tsx

```diff
@@ -93,6 +98,33 @@ export function Notifications({ iconSize = 24 }: { iconSize?: number }) {
 
   const ref = useRefClickOutside<HTMLDivElement>({ open, setOpen });
 
+  function onIconClick() {
+    setOpen(!open);
+
+    if (!open && notifications.length) {
+      const latestNotificationTime = notifications[0].createdDate;
+      // If the current userLastSeenNotification is already set to the most recent notification's time, don't do anything.
+      if (userLastSeenNotification === latestNotificationTime) {
+        return;
+      }
+
+      const previousLastSeenTime = userLastSeenNotification;
+
+      // Update the visual state right away for a snappier UX.
+      setUserLastSeenNotification(latestNotificationTime);
```

Comment:
I have troubles understanding what happens if we don't set this, why is it snappier?

Reply:
Because otherwise we would need to wait for the response to be returned before the visual state is updated, so if let's say the request took a full second, the user would open the notifications pane and the `9+` badge will remain visible for a full second.